.github/actions/link-package/action.yml

```diff
@@ -26,29 +27,44 @@ inputs:
 runs:
   using: composite
   steps:
-    - name: Checkout package repo 🛎
+    - name: Checkout external package repo 🛎
       uses: actions/checkout@v3
       with:
         repository: ${{ inputs.external_pkg_repo }}
         ref: ${{ inputs.external_pkg_ref }}
         path: external-pkg
 
-    # following https://github.com/jupyterlab/lumino/blob/main/CONTRIBUTING.md#setting-up-a-development-environment
-    # and https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#id17
-    - name: Build package from source 🏗
+    # https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#id18
+    - name: Build external package from source 🏗
+      id: build-pkg
       run: |
         echo "::group::Build package from source 🛠"
         conda run --prefix $CONDA_PREFIX yarn
         conda run --prefix $CONDA_PREFIX yarn build:src
-        conda run --prefix $CONDA_PREFIX YARN_IGNORE_PATH=1 yarn link
         echo "::endgroup::"
+        echo $(pwd)
+        sudo apt install -y jq
+        echo "pkg-name=$(jq '.name' package.json)" >> $GITHUB_OUTPUT
       shell: bash -l {0}
       working-directory: external-pkg
 
-    - name: Link JupyterLab and package 🔗
+    - name: Link JupyterLab and external package 🔗
       run: |
-        # todo later on: replace the lumino reference here and output linked packages
-        conda run --prefix $CONDA_PREFIX jlpm link "lumino-top-level"
+        conda run --prefix $CONDA_PREFIX yarn link "${{ github.workspace }}/external-pkg" -p
```

Comment:
Ahh, so you never really needed to grab the package name because the new version of Yarn just lets you [link to the path where the package is installed](https://yarnpkg.com/cli/link). 

This is great!

Reply:
yup this was much neater